### PR TITLE
Nginx Stub Status Receiver

### PR DESCRIPTION
**Pull request by keitwb**
This is a receiver for the Nginx stub_status module output

It collects all of the metrics available from that module.

### CODE REVIEW

File: receiver/nginxreceiver/config.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package nginxreceiver
+
+import (
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+type config struct {
+	configmodels.ReceiverSettings            `mapstructure:",squash"`
```
Comment (bogdandrutu) [inline receiver/nginxreceiver/config.go:24]: This is not needed, the ScraperControllerSettings includes that.

File: receiver/nginxreceiver/internal/metadata/.nocover [hunk 2]
```diff
@@ -0,0 +1 @@
+We test the generated code in the package it is used.
```
Comment (bogdandrutu) [inline receiver/nginxreceiver/internal/metadata/.nocover:1]: No need for this. Also it was removed as a requirement from core as well.

**Review by bogdandrutu — COMMENTED** [review]
Why is there an empty Dockerfile?

**Comment by keitwb** [comment]
> Why is there an empty Dockerfile?

That was a leftover of trying to do an integration test but I realized it is much easier and just as robust to do a unit test with httptest. Removed.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**